muskrat/s3consumer.py

```diff
@@ -9,10 +9,54 @@
 """
 import os
 import time
+import json
 
-import boto
+import boto3
 from   muskrat.util import config_loader
 
+
+def s3collection_marker_each_cur(collection, cursor):
+    """
+    Helper for iterating through s3 objects using a persistent marker. Loads
+    the marker from the given file when present, and creates it if not present.
+
+    collection
+        a collection of boto3 s3 ObjectSummary resources
+    cursor
+        S3Cursor instance where the marker will be read and stored
+    """
+    marker = cursor.get()
+    if marker:
+        collection = collection.filter(Marker=marker)
+
+    for obj in collection:
+        yield obj
```

Comment:
As long as we're working on Muskrat it would be nice to feed optional metadata into the consumer.  Namely, the cursor is not available to the consumer except by inspecting the consumer property that is added to the function.  Even then it only corresponds to the last cursor that was written.  Would definitely be nice to have this in a few cases.  The main one is, of course, logging.